src/backends/github/API.js

```diff
@@ -20,70 +21,44 @@ export default class API {
   }
 
   isCollaborator(user) {
-    return this.request('/user/repos').then((repos) => {
-      let contributor = false
-      for (const repo of repos) {
-        if (repo.full_name === this.repo && repo.permissions.push) contributor = true;
-      }
-      return contributor;
-    }).catch((error) => {
-      console.error("Problem with response of /user/repos from GitHub");
-      throw error;
-    })
+    return this.request('/user/repos').then(repos =>
+      Object.keys(repos).some(
+        key => (repos[key].full_name === this.repo) && repos[key].permissions.push
+      )
+    );
   }
 
   requestHeaders(headers = {}) {
-    const baseHeader = {
-      "Content-Type": "application/json",
+    return {
       ...headers,
+      ...(this.token ? { Authorization: `token ${ this.token }` } : {}),
```

Comment:
is there a reason to include the empty `{}` when there's no token as opposed to using
`...(this.token && { Authorization: `token ${ this.token}` }),`? I would think unless something is looking for data based on array position, removing an empty object would be preferable.